subsys/bluetooth/controller/hci/hci.c

```diff
@@ -3564,6 +3565,37 @@ static void le_adv_ext_coded_report(struct pdu_data *pdu_data,
 {
 	le_adv_ext_report(pdu_data, node_rx, buf, BIT(2));
 }
+
+static void le_adv_ext_terminate(struct pdu_data *pdu_data,
+				    struct node_rx_pdu *node_rx,
+				    struct net_buf *buf)
+{
+	struct bt_hci_evt_le_adv_set_terminated *sep;
+
+	if (!(event_mask & BT_EVT_MASK_LE_META_EVENT) ||
+	    !(le_event_mask & BT_EVT_MASK_LE_ADV_SET_TERMINATED)) {
+		return;
+	}
+
+	sep = meta_evt(buf, BT_HCI_EVT_LE_ADV_SET_TERMINATED, sizeof(*sep));
+
+	/* Bit 17 set indicates event was caused by connection */
+	/* Bit 16 set indicates event was caused by duration timeout */
+	if ((uint32_t)node_rx->hdr.rx_ftr.extra & BIT(TERM_EVT_CONNECT)) {
```

Comment:
Return the status in bits 16-23 directly instead of using bit fields?

Reply:
Where do I send status? node_rx_hdr has extra and param fields, which we use for # of elapsed events and conn handle. Since we only use 16 out of 32 bits in extra, that seemed like a good place for these bits, which determine status. But we could also put status (8 bits) directly in bits 16-23 of extra.

Reply:
Added status as bit 8-15 in extra.